src/ProductManifold.jl

```diff
@@ -178,37 +178,55 @@ end
 
 
 """
-    is_manifold_point(M::ProductManifold, x; kwargs...)
+    manifold_point_error(M::ProductManifold, x; kwargs...)
 
 Check whether `x` is a valid point on the [`ProductManifold`](@ref) `M`.
 
 The tolerance for the last test can be set using the ´kwargs...`.
 """
-function is_manifold_point(M::ProductManifold, x::ProductRepr; kwargs...)
-    return all(t -> is_manifold_point(t...; kwargs...), ziptuples(M.manifolds, x.parts))
+function manifold_point_error(M::ProductManifold, x::ProductRepr; kwargs...)
+    for t ∈ ziptuples(M.manifolds, x.parts)
+        err = manifold_point_error(t...; kwargs...)
+        err === nothing || return err
```

Comment:
This would just return the first error occuring, right?

If one works on code and looks for errors that would mean one has to solve one error then gets the second, solves that ...
Is there a way to collect all errors and return a string containing all messages?

I think I would find that better to get all errors at once (let's say if you made a mistake in 4 components).

Reply:
Right, only the user is informed only about the first returned error. Your suggestion is good but that requires some design changes. Can we assume that `manifold_point_error` always returns a string or `nothing` and `check_manifold_point` throws a `DomainError` if `manifold_point_error` returns something else than `nothing`? Or maybe we could even return an empty string from `manifold_point_error` if there is no error (instead of `nothing`).

Reply:
Well one could leave that at nothing, collect an array of the inner `manifold_point_errors()`, filter out nothings and return nothing if the filtered result is empty. Otherwise join the strings (error messages), maybe with a `"\n"` in between. Since here we always only do strings, the check anyways has no choice than return a `DomainError`. What do you think about that?

Reply:
That looks fine, I'll do it.

Reply:
There is a small problem with this, `DomainError` should carry two things: the problematic value and an (optional) error message. If we merge two error messages, the value would have to be discarded. The current solution is not worse than what we had so I'd just open an issue about it and solve it later.